File: backend-dotnet/Domain/Entities/PostTag.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class PostTag
    {
        [Required]
        public Guid PostId { get; set; }
        public Post Post { get; set; } = default!;
        [Required]
        public Guid TagId { get; set; }
        public Tag Tag { get; set; } = default!;
    }
}

File: backend-java/src/main/java/com/jerrygram/domain/entities/PostTag.java
package com.jerrygram.domain.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "\"PostTags\"")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@IdClass(PostTagId.class)
public class PostTag {
    
    @Id
    @Column(name = "\"PostId\"", nullable = false)
    private UUID postId;
    
    @Id
    @Column(name = "\"TagId\"", nullable = false)
    private UUID tagId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"PostId\"", nullable = false, insertable = false, updatable = false)
    private Post post;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"TagId\"", nullable = false, insertable = false, updatable = false)
    private Tag tag;
}